doc/sphinx/proof-engine/tactics.rst

```diff
@@ -264,29 +267,31 @@ Applying theorems
 .. tacn:: apply @term
    :name: apply
 
-   This tactic applies to any goal. The argument term is a term well-formed in the
-   local context. The tactic apply tries to match the current goal against the
-   conclusion of the type of term. If it succeeds, then the tactic returns as many
-   subgoals as the number of non-dependent premises of the type of term. If the
-   conclusion of the type of term does not match the goal *and* the conclusion is
-   an inductive type isomorphic to a tuple type, then each component of the tuple
-   is recursively matched to the goal in the left-to-right order.
-
-   The tactic :tacn:`apply` relies on first-order unification with dependent types
-   unless the conclusion of the type of :token:`term` is of the form :g:`P (t`:sub:`1`
-   :g:`...` :g:`t`:sub:`n` :g:`)` with `P` to be instantiated. In the latter case, the behavior
-   depends on the form of the goal. If the goal is of the form
-   :g:`(fun x => Q) u`:sub:`1` :g:`...` :g:`u`:sub:`n` and the :g:`t`:sub:`i` and
-   :g:`u`:sub:`i` unifies, then :g:`P` is taken to be :g:`(fun x => Q)`. Otherwise,
-   :tacn:`apply` tries to define :g:`P` by abstracting over :g:`t`:sub:`1`  :g:`...`
-   :g:`t`:sub:`n` in the goal. See :tacn:`pattern` to transform the goal so that it
-   gets the form :g:`(fun x => Q) u`:sub:`1` :g:`...`  :g:`u`:sub:`n`.
+   This tactic applies to any goal. The argument term is a term well-formed in
+   the local context. The tactic :tacn:`apply` tries to match the current goal
+   against the conclusion of the type of :token:`term`. If it succeeds, then
+   the tactic returns as many subgoals as the number of non-dependent premises
+   of the type of term. If the conclusion of the type of :token:`term` does
+   not match the goal *and* the conclusion is an inductive type isomorphic to
+   a tuple type, then each component of the tuple is recursively matched to
+   the goal in the left-to-right order.
+
+   The tactic :tacn:`apply` relies on first-order unification with dependent
+   types unless the conclusion of the type of :token:`term` is of the form
+   :n:`P (t__1 ... t__n)` with ``P`` to be instantiated. In the latter case,
```

Comment:
Should this be `P ({+ t})`? (I think the way you wrote it is fine, too; just wondering)

Reply:
I think I kept it that way because of the later question of whether `t__i` and `u__i` unify.